### DISCUSSION TITLE
Moved to a new server, every service works except mailserver

### DISCUSSION THREAD

**Discussion by aario** [docker-mailserver/docker-mailserver#3851] [Q&A]
Hi Folks,
Today I voved to a new server, every service works except mailserver.
Telnet can make a connection to 993. If I stop the container, it cannot. So obviously the connection has been established.
Two different mail clients, k9mail of android and kmail of linux, both could connect on the previous server. Now they timeout and go offline mode.
There is no error on the log of dovecot. Nothing.
Anything else to check? Please help. Thanks in advance.

```
root@mail:/# doveconf listen protocols
listen = *, ::
protocols =  imap lmtp
root@mail:/# ps auxf | grep dovecot
root        4816  0.0  0.0   2904   676 pts/0    S+   20:39   0:00  \_ grep dovecot
root         651  0.0  0.0   4304  2844 ?        S    19:56   0:00  \_ /usr/sbin/dovecot -F -c /etc/dovecot/dovecot.conf
dovecot      656  0.0  0.0   3860   840 ?        S    19:56   0:00  |   \_ dovecot/anvil
root         657  0.0  0.0   3992  2448 ?        S    19:56   0:00  |   \_ dovecot/log
root         658  0.0  0.1   6888  4164 ?        S    19:56   0:00  |   \_ dovecot/config
root@mail:/# netstat -ptan | grep ':993'
tcp        0      0 0.0.0.0:993             0.0.0.0:*               LISTEN      651/dovecot         
tcp6       0      0 :::993                  :::*                    LISTEN      651/dovecot         
```

**Comment by polarathene** [comment]
You have shared very little about your config.

Can you setup a new DMS instance on that server, or temporarily disable your volume mounts. Does it work now?

If not, can you remove any extra config like ENV, so that it's just the ports published and hostname configured (_`docker exec -it dms-container setup email add test@your-hostname bad-password` to create a temporary account_), any issues now?

**Reply by polarathene** [reply]
Also, what changed between the two servers? OS? Packages? Something about your environment may also have changed.

Were you using a pinned version in your config? Or relying on `:latest`? (_which is very bad_) If you did not pin your version, you may have accidentally upgraded to a new major release with breaking changes, which broke part of your setup.

**Reply by aario** [reply]
Thanks for the reply. Actually I found that running `strace -p` you can attach to the running process to find out what is he waiting for all the time!
For me it was waiting for `epoll_wait`. That made me think it is a problem with file permissions mounted into the container:
https://serverfault.com/questions/500980/dovecot-hangs-on-all-logins-including-local-telnet
I actually did find some folders which are owned by the root on the old server and by the normal user on the new server. So those can be a problem and I"m currently getting them fixed. Will let you know if that worked.

**Reply by polarathene** [reply]
> There is no error on the log of dovecot. Nothing.

What about your log from DMS?

- Have you looked at [our troubleshooting docs](https://docker-mailserver.github.io/docker-mailserver/v13.3/config/debugging/)?
- Set the [`LOG_LEVEL` ENV](https://docker-mailserver.github.io/docker-mailserver/v13.3/config/environment/#log_level) to a more verbose level like `debug`, then check the `docker logs` output for anything that looks odd / failing / error / timeout.

**Reply by aario** [reply]
So I ensured every file permission is as in the old server. Modified date of files shows that they are being accessed by the container. But still does not work. I increased the LOG_LEVEL to debug. Could not find any error. Here is the log:
```
[  DEBUG  ]  Handling general environment variable setup
[   INF   ]  Welcome to docker-mailserver v13.3.1
[  DEBUG  ]  Registering functions
[   INF   ]  Checking configuration
[  DEBUG  ]  Checking for improper restart
[  DEBUG  ]  Checking that hostname/domainname is provided or overridden
[  DEBUG  ]  Domain has been set to masked-server-address.com
[  DEBUG  ]  Hostname has been set to mail.masked-server-address.com
[   INF   ]  Configuring mail server
[  DEBUG  ]  Setting up general log files
[  DEBUG  ]  Setting up Dovecot
[  DEBUG  ]  Enabling IMAP services
[  DEBUG  ]  Setting up Dovecot dhparam
[  DEBUG  ]  Setting up Dovecot quota
[  DEBUG  ]  Spam emails will be moved to the Junk folder
[  DEBUG  ]  Spam emails will not be marked as read
[  DEBUG  ]  Setting up Dovecot Local User
[  DEBUG  ]  Creating user 'me' for domain 'masked-server-address.com'
[  DEBUG  ]  Creating user 'vendors' for domain 'masked-server-address.com'
[  DEBUG  ]  Creating user 'cloud' for domain 'masked-server-address.net'
[  DEBUG  ]  Configuring DKIM
[  DEBUG  ]  Configuring OpenDMARC
[  DEBUG  ]  Configuring policyd-spf
[  DEBUG  ]  Setting up Security Stack
[  DEBUG  ]  Enabling and configuring Postgrey
[  DEBUG  ]  Configuring Postscreen
[  DEBUG  ]  Disabling Postscreen DNSBLs
[  DEBUG  ]  Enabling and configuring SpamAssassin
[  DEBUG  ]  'SPAMASSASSIN_SPAM_TO_INBOX=1' is set. The 'SA_KILL' ENV will be ignored.
[  DEBUG  ]  Disabling ClamAV
[  DEBUG  ]  Enabling and configuring Fail2Ban
[  DEBUG  ]  Configuring Amavis
[  DEBUG  ]  Rspamd is disabled
[  DEBUG  ]  Setting up SSL
[  DEBUG  ]  TLS configured with 'modern' ciphers
[  DEBUG  ]  Configuring SSL using 'letsencrypt'
[  DEBUG  ]  Setting up PERMIT_DOCKER option
[  DEBUG  ]  Setting up mailname and creating '/etc/mailname'
[  DEBUG  ]  Applying hostname to Dovecot
[  DEBUG  ]  Configuring Postfix (early setup)
[  DEBUG  ]  '/tmp/docker-mailserver/postfix-virtual.cf' not provided - no mail alias/forward created
[  DEBUG  ]  Setting up Postfix dhparam
[  DEBUG  ]  Spoof protection is disabled
[  DEBUG  ]  Configuring Postfix (late setup)
[  DEBUG  ]  Setting up Postfix Relay Hosts
[  DEBUG  ]  (Postfix setup) Overriding / adjusting configuration with user-supplied values
[  DEBUG  ]  Fetchmail is disabled
[  DEBUG  ]  Fetchmail parallel is disabled
[  DEBUG  ]  Getmail is disabled
[  DEBUG  ]  Setting up logrotate
[  DEBUG  ]  Postfix log summary reports disabled
[  DEBUG  ]  Logwatch reports disabled.
[  DEBUG  ]  Consolidating all state onto /var/mail-state
[  DEBUG  ]  Checking /var/mail permissions
[  DEBUG  ]  Removing files and directories from older versions
[  DEBUG  ]  Ensuring  is owned by '_rspamd:_rspamd'
chown: cannot access '': No such file or directory
[  DEBUG  ]  Exporting environment variables now (creating '/etc/dms-settings')
[  DEBUG  ]  Setting up configuration checksum file
[  DEBUG  ]  Printing environment variables. Make sure no sensitive data is copied.
ACCOUNT_PROVISIONER='FILE'
AMAVIS_LOGLEVEL='0'
CLAMAV_MESSAGE_SIZE_LIMIT='25M'
DEFAULT_RELAY_HOST=''
DMS_VMAIL_GID='5000'
DMS_VMAIL_UID='5000'
DOVECOT_INET_PROTOCOLS='all'
DOVECOT_MAILBOX_FORMAT='maildir'
DOVECOT_TLS='no'
ENABLE_AMAVIS='1'
ENABLE_CLAMAV='0'
ENABLE_DNSBL='0'
ENABLE_FAIL2BAN='1'
ENABLE_FETCHMAIL='0'
ENABLE_GETMAIL='0'
ENABLE_IMAP='1'
ENABLE_MANAGESIEVE='0'
ENABLE_OAUTH2='0'
ENABLE_OPENDKIM='1'
ENABLE_OPENDMARC='1'
ENABLE_POLICYD_SPF='1'
ENABLE_POP3='0'
ENABLE_POSTGREY='1'
ENABLE_QUOTAS='1'
ENABLE_RSPAMD='0'
ENABLE_RSPAMD_REDIS='0'
ENABLE_SASLAUTHD='0'
ENABLE_SPAMASSASSIN='1'
ENABLE_SPAMASSASSIN_KAM='0'
ENABLE_SRS='0'
ENABLE_UPDATE_CHECK='1'
FAIL2BAN_BLOCKTYPE='drop'
FETCHMAIL_PARALLEL='0'
FETCHMAIL_POLL='300'
GETMAIL_POLL='5'
LOGROTATE_INTERVAL='weekly'
LOGWATCH_INTERVAL='none'
LOGWATCH_RECIPIENT='postmaster@masked-server-address.com'
LOGWATCH_SENDER='mailserver-report@mail.masked-server-address.com'
LOG_LEVEL='debug'
MARK_SPAM_AS_READ='0'
MOVE_SPAM_TO_JUNK='1'
NETWORK_INTERFACE='eth0'
ONE_DIR='1'
OVERRIDE_HOSTNAME=''
PERMIT_DOCKER='none'
PFLOGSUMM_RECIPIENT='postmaster@masked-server-address.com'
PFLOGSUMM_SENDER='mailserver-report@mail.masked-server-address.com'
PFLOGSUMM_TRIGGER='none'
POSTFIX_DAGENT=''
POSTFIX_INET_PROTOCOLS='all'
POSTFIX_MAILBOX_SIZE_LIMIT='0'
POSTFIX_MESSAGE_SIZE_LIMIT='10240000'
POSTFIX_REJECT_UNKNOWN_CLIENT_HOSTNAME='0'
POSTGREY_AUTO_WHITELIST_CLIENTS='5'
POSTGREY_DELAY='300'
POSTGREY_MAX_AGE='35'
POSTGREY_TEXT='Delayed by Postgrey'
POSTMASTER_ADDRESS='postmaster@masked-server-address.com'
POSTSCREEN_ACTION='enforce'
RELAY_HOST=''
REPORT_RECIPIENT='postmaster@masked-server-address.com'
REPORT_SENDER='mailserver-report@mail.masked-server-address.com'
RSPAMD_CHECK_AUTHENTICATED='0'
RSPAMD_GREYLISTING='0'
RSPAMD_HFILTER='1'
RSPAMD_HFILTER_HOSTNAME_UNKNOWN_SCORE='6'
RSPAMD_LEARN='0'
SA_KILL='10.0'
SA_SPAM_SUBJECT='***SPAM*** '
SA_TAG2='6.31'
SA_TAG='2.0'
SMTP_ONLY='0'
SPAMASSASSIN_SPAM_TO_INBOX='1'
SPOOF_PROTECTION='0'
SRS_DOMAINNAME='masked-server-address.com'
SRS_EXCLUDE_DOMAINS=''
SRS_SECRET=''
SRS_SENDER_CLASSES='envelope_sender'
SSL_TYPE='letsencrypt'
SUPERVISOR_LOGLEVEL='warn'
TLS_LEVEL='modern'
TZ=''
UPDATE_CHECK_INTERVAL='1d'
VIRUSMAILS_DELETE_DELAY='7'
[   INF   ]  Starting daemons
[  DEBUG  ]  Starting cron
[  DEBUG  ]  Starting rsyslog
[  DEBUG  ]  Starting dovecot
[  DEBUG  ]  Starting update-check
[  DEBUG  ]  Starting opendkim
[  DEBUG  ]  Starting opendmarc
[  DEBUG  ]  Starting postgrey
[  DEBUG  ]  Starting postfix
[  DEBUG  ]  Starting fail2ban
[  DEBUG  ]  Starting amavis
[  DEBUG  ]  Starting changedetector
[   INF   ]  mail.masked-server-address.com is up and running
Jan 29 22:44:23 mail amavis[830]: starting. /usr/sbin/amavisd-new at mail.masked-server-address.com amavisd-new-2.11.1 (20181009), Unicode aware, LC_CTYPE="C.UTF-8"
Jan 29 22:44:23 mail amavis[830]: perl=5.032001, user=, EUID: 110 (110);  group=, EGID: 112 112 (112 112)
Jan 29 22:44:24 mail amavis[830]: Net::Server: Group Not Defined.  Defaulting to EGID '112 112'
Jan 29 22:44:24 mail amavis[830]: Net::Server: User Not Defined.  Defaulting to EUID '110'
Jan 29 22:44:24 mail amavis[830]: No ext program for   .zoo, tried: zoo
Jan 29 22:44:24 mail amavis[830]: No ext program for   .doc, tried: ripole
Jan 29 22:44:24 mail amavis[830]: No decoder for       .F   
Jan 29 22:44:24 mail amavis[830]: No decoder for       .doc 
Jan 29 22:44:24 mail amavis[830]: No decoder for       .zoo 
```

**Reply by polarathene** [reply]
The log output only shows that DMS is up and running.

There should be some other logs when you attempt to connect and it stalls. Presently the logs look fine.

Can you please try without any existing volumes? It's possible that they contribute to the issue. You need to prioritize the minimal config example like I showed, and that it is successful at establishing a connection where your real config isn't.

If DMS still fails to be successful in the minimal config example, then it's not a DMS issue but something else about your new server setup for you to look into.

**Comment by casperklein** [comment]
> Now they timeout and go offline mode. There is no error on the log of dovecot. Nothing.

- Did you update your DNS records to point to the new server?
- Check fail2ban if used: `./setup.sh fail2ban`
- Check host firewall?
- Check if network packets arrive at your server: `tcpdump port 993`

**Reply by aario** [reply]
As for network connections, if I stop dovecot, I cannot telnet to it. If I don't and let it run, I can telnet. So it behaves like a cat! It listens. But does not feel like answering!

**Reply by casperklein** [reply]
> if I stop dovecot, I cannot telnet to it

Do you test telnet and k9mail on the same machine? If you get different results, then it's indeed weird.

**Reply by aario** [reply]
It is not possible to run the clients on the server. Specially that k9mail is android app and I cannot run it on the server. So they are run on different devices.

**Reply by casperklein** [reply]
That was my point. When you locally on the host run telnet and that works, but k9 on an other client can't even connect, check fail2ban etc.

**Reply by aario** [reply]
Telnet works on my laptop and on local of the server. In both cases connection is established. But the server does not reply.
If I stop the container, connection cannot be established.

**Reply by polarathene** [reply]
If you need assistance you need to follow the instructions I gave earlier.

Ignore your current DMS config, verify if a minimal setup works or still fails since this is a new server.

```console
# Basic DMS container (_it will expect a user added within 2 minutes of starting or it will shutdown_):
docker run --rm -it --hostname mail.example.test -p 143 -p 587 -e LOG_LEVEL=debug --name dms mailserver/docker-mailserver

# Add a user:
docker exec -it dms setup email add user@example.test bad-password

# Send a test mail:
docker exec -it dms swaks --server localhost:587 --to user@example.test --auth-user user@example.test --auth-password bad-password

# Connect to Dovecot IMAP if you want, try check the new mail
# Check logs for any issues:
docker logs dms
```

None of that requires DNS setup. Thunderbird can connect to it locally with config similar to this image:

![image](https://github.com/docker-mailserver/docker-mailserver/assets/5098581/d26c6d7e-0824-4a29-9b0f-395b91aa7603)

Local host firewall could be an issue though among other things, so it's best to verify minimum reproduction.

---

If the above doesn't have an issue, you can add your ENV config, then your volumes, then test with actual DNS, etc until it breaks.

**Reply by aario** [reply]
Oh no!!!! The `setup fail2ban` command just showed me that my IP has been banned on my own server! Thanks man. You saved my day!

**Reply by polarathene** [reply]
Since the server is headless, you'd perhaps want to install a CLI IMAP client instead to test on the server that mail retrieval works there.

You could look into `mutt` / `alpine` / `himalaya`..

---

You did bring up `telnet`, but not sharing much on how you did that, did you try with `localhost` to ignore the DNS concerns?

You can try IMAP login verification with curl:

```bash
curl --silent --verbose \
    --login-options 'AUTH=PLAIN' --user 'user@example.test:bad-password' \
    --url 'imap://localhost:143' -X 'LOGOUT'
```

Perhaps the verbose output provides some insights?

**Reply by polarathene** [reply]
Nevermind my replies, glad you got it resolved!

Github wasn't updating the comments until I sent mine and a bunch of new ones on this thread were shown 😫

**Reply by aario** [reply]
Now it's too soon to celebrate. Cause I have unbanned my IP from fail2banl. Checked the status. It's not banned anymore. Restarted the containers. Still mail server does not work. Anywhere else to look?

**Reply by aario** [reply]
As you said in your comment, ran:
```
curl --silent --verbose \
    --login-options 'AUTH=PLAIN' --user 'user@example.test:bad-password' \
    --url 'imap://localhost:143' -X 'LOGOUT'
```
With proper username and password and here is the log:
```
    Jan 29 23:12:25 mail dovecot: imap-login: Login: user=<my username>, method=PLAIN, rip=127.0.0.1, lip=127.0.0.1, mpid=1554, secured, session=<sessionid>
Jan 29 23:12:25 mail dovecot: imap(my username)<1554><sessionid>: Logged out in=11 out=485 deleted=0 expunged=0 trashed=0 hdr_count=0 hdr_bytes=0 body_count=0 body_bytes=0

```

**Reply by polarathene** [reply]
> here is the log

That looks fine then.

You can use curl to do more IMAP commands but it's a bit awkward I think.

Try run a local IMAP client on the server like I suggested, if that can access mail and the test mail is there then there is no problem with DMS AFAIK. It'll be with your server, probably a firewall?

**Reply by aario** [reply]
Interesting: Inside container curl works. Outside container and on the server, it does not. It just hangs there.

**Reply by aario** [reply]
Various IPs been banned. Some container related IPs. Some docker related IPs. Some server related IPs. I ran:
```
fail2ban-client unban --all
```
From inside the container and the issue got fixed.
The command `iptables -L` is useless, as for some unknown reason, it does not show fail2ban IPs.
So, the issue is fixed now. Thanks everyone.
